Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -83,6 +90,12 @@ private void handleBraintreeEventBundle(Bundle bundle) {
             VaultedPaymentMethodSelectedEvent vaultedPaymentMethodSelectedEvent =
                     (VaultedPaymentMethodSelectedEvent) braintreeResult;
             onVaultedPaymentMethodSelected(vaultedPaymentMethodSelectedEvent.getPaymentMethodNonce());
+        } else if (braintreeResult instanceof AddCardEvent) {
+            showCardDetailsFragment(((AddCardEvent) braintreeResult).getCardNumber());
+        } else if (braintreeResult instanceof CardDetailsEvent) {
+            onCardDetailsEvent((CardDetailsEvent) braintreeResult);
+        } else if (braintreeResult instanceof EditCardNumberEvent) {
+            startAddCardFlow((EditCardNumberEvent) braintreeResult);
```

Comment:
The naming here for these `Event` classes threw me off a bit. After stepping through the flow on a simulator, it makes more sense:`CardDetailsEvent` represents _after_ card details have been collected, `AddCardEvent` represents _after_ card number is collected. 

Is this Android convention? If not, is there a way we could make this a bit more clear?

Reply:
We matched Event and Fragment prefixes, e.g. `AddCardEvent` and `AddCardFragment` etc, mainly to call out the origin of the event. Another option would be to name the event based on the action that should be taken. I'm up for changing this though in the long run.

Reply:
Once all the activities are gone we should review the overall structure of DropIn. A big goal with the events is to decouple the host activity from the fragments and make each component easier to test.

Reply:
Going to add a TODO here. We may be able to create a single `AppEvent` type to simplify everything.